### PR TITLE
[overlay] add clickable overlay for launching gui/launcher

### PR DESCRIPTION
**Pull request by myk002**
image
image

### CODE REVIEW

File: plugins/overlay.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+struct title_hook : df::viewscreen_titlest {
+    typedef df::viewscreen_titlest interpose_base;
+
+    DEFINE_VMETHOD_INTERPOSE(void, feed, (set<df::interface_key> *input)) {
+        if (!handle_click())
+            INTERPOSE_NEXT(feed)(input);
+    }
+    DEFINE_VMETHOD_INTERPOSE(void, render, ()) {
+        INTERPOSE_NEXT(render)();
+        draw_widgets();
```
Comment (lethosor) [inline plugins/overlay.cpp:47]: I would swap the order of these, in case screens are drawing something important in this region

Comment (myk002) [reply plugins/overlay.cpp:47]: I tested that, and nothing is shown at all if the order is reversed. At least on the title screen.

Comment (lethosor) [reply plugins/overlay.cpp:47]: Bah, it must be drawing blank characters on the whole screen. Any screen using drawBorder() would run into the same thing, probably.

Not sure what a good fix is, then. Maybe scanning for text characters before writing? I don't want to overwrite things like the weather indicator.

Comment (myk002) [reply plugins/overlay.cpp:47]: I planned to move the weather indicator somewhere non-conflicting, like a few tiles to the right. Are there others that I need to watch out for?

Comment (myk002) [reply plugins/overlay.cpp:131]: done, but you have to `rm dfhack-config/dwarfmonitor.json` to get the update to the weather indicator position.

Comment (lethosor) [reply plugins/overlay.cpp:131]: I guess moving that is fine. I noticed the controls on `gui/load-screen` when I was loading a game. That one won't be covered if we are skipping DFHack screens, though.

I don't know of any DF screens that would conflict off the top of my head - most draw a border, which we can draw on top of easily enough. Exceptions that I know of: title, options, save, load, clean

**Comment by myk002** [comment]
merging so TWBT can make progress. more work is needed on our side, but basic mouse grid mapping is tested to work

**Pull request merged**